capabilities/navigation/nearby-poi/maps/google-apis-maps.suma

```diff
@@ -0,0 +1,108 @@
+profile = "navigation/nearby-poi@1.0"
+provider = "google-apis-maps"
+
+map NearbyPoi {
+  categories = input.categories || []
+
+  http GET "/maps/api/place/nearbysearch/json" {
+    security "apikey"
+
+    request {
+      query {
+        location = `${input.center.latitude},${input.center.longitude}`,
+        radius = input.radius,
+        type = call BuildCategoryQueryType(categories = categories)
+      }
+    }
+
+    response 200 "application/json" {
+      return map result if (body.status === "ZERO_RESULTS") []
+
+      return map error if (body.status !== "OK") {
+        status: body.status,
+        message: body.error_message || body.status
+      }
+
+      points = call foreach(place of body.results) ConvertResponse(place = place)
+      set if (categories.length > 1) {
+        points = points.filter(
+          p => p.categories.some(c => categories.includes(c))
+        )
+      }
+
+      map result points
+    }
+  }
+}
+
+operation BuildCategoryQueryType {
+  return if (args.categories.length !== 1) undefined
```

Comment:
This will return undefined when there will be more than one categories, is it intentional?

```suggestion
  return if (args.categories.length === 0) undefined
```

Reply:
Sorry, I forgot to comment this. It returns `undefined` for anything but one category because google apis don't allow specifying multiple categories.

When the user requests one category, the map should propagate that to the API. When the user requests zero, two or more categories, we should not pass any category filter to the API, otherwise they would only get the first category they wanted and no other.

Reply:
Oh right, categories are passed as array, so they're formated like `param,param`. Would formatting categories like `param|param` work? Based on their [docs](https://developers.google.com/maps/documentation/places/web-service/search-nearby#type) it takes the first specified, but still better than no category. 🤷🏻‍♂️

Reply:
I don't think it's better, because then it will never return any category other than the first, which directly contradicts the idea of requesting multiple categories.

I choose this because it will return all POIs in the vicinity and then there is a filter [here](https://github.com/superfaceai/station/pull/66/files/f40beb2a0762c2ebd0adec3c5e93299f5f490a00#diff-657a37b57be357898afc628af21da103e7e79f4a2c1b053cabcc0b1ae4ee17e7R28) that makes sure only the specified categories are actually returned by the map.

The big downside of this is that the accuracy will be severely limited since the API returns many other categories so the selected ones will be spread out.

My only other idea was to run the request for each category, but that might be an unexpected cost to the user because it will fire off several requests, each of which will be billed by Google.

Reply:
Oh nice, I didn't understood that filter. This is definitely better.

Reply:
@Jakub-Vacek can you please revert suggested change?